### ISSUE TITLE
Clicking Take me there doesn't automatically sign user  in to New Expensify

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#53034]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.66-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation** (hyperlinked to channel name): [Expensify-migrate](https://expensify.slack.com/archives/C07NMDKEFMH/p1732311983432619)

## Action Performed:
1. Submit a report to a workspace approver
2. In OD open the report
3. Click on "take me there"
## Expected Result:
User logged in to ND with automatically signed in 
## Actual Result:
User is taken to login screen
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

https://github.com/user-attachments/assets/d0833a5e-f68f-49ae-aaac-b0a591300fa4

https://github.com/user-attachments/assets/5637799d-c58a-442c-98b5-4f752965d860

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by joekaufmanexpensify** [comment]
Reproduced.

**Comment by klajdipaja** [comment]
This is an issue on the Old Dot. 
`Take me there` button is pointing to the wrong environment url, it should point to `staging.expensify.com` instead it's point to `new.expensify.com/expenses`

**Comment by joekaufmanexpensify** [comment]
@rojiphil Are you seeing somewhere that it's expected with the current design that we don't support the case of transferring you from OldDot > NewDot without explicitly signing into NewDot?

We definitely support the reverse now, so I'd think we'd support this too.

**Comment by rojiphil** [comment]
> @rojiphil Are you seeing somewhere that it's expected with the current design that we don't support the case of transferring you from OldDot > NewDot without explicitly signing into NewDot?
> 
> We definitely support the reverse now, so I'd think we'd support this too.

Yeah. I too agree that the transferring from OD > ND should be supported. I was just confirming if this was previously implemented or is a new feature. 
Anyway I think we need BE support as OD must invoke the URL with auth credentials as params following which we can set the session data locally to automatically sign in. 
Trigger the involvement of an internal engineer here to take this forward.
 🎀👀🎀

**Comment by rojiphil** [comment]
> But I think we need BE support as OD must invoke the URL with auth credentials as params following which we can set the session data locally to automatically signin. 

@marcaaron But there are also references to signing in from OD [here](https://github.com/Expensify/App/blob/9ea0f51cba69f6183386eeb39f5f751e77968271/src/components/InitialURLContextProvider.tsx#L54). So, curious to know if this feature is supposed to work with Hybrid Apps. What do you think?

**Comment by marcaaron** [comment]
>I was just confirming if this was previously implemented or is a new feature.

I think this is a good distinction. @joekaufmanexpensify maybe you can help figure out who has the most context around this new feature (or is it a bug)? It will be some time before I can look into this. I see it's labeled as a "HIGH" in the #migrate project (maybe someone has bandwidth).

**Comment by rojiphil** [comment]
> > I was just confirming if this was previously implemented or is a new feature.
> 
> I think this is a good distinction. @joekaufmanexpensify maybe you can help figure out who has the most context around this new feature (or is it a bug)?

Thanks @marcaaron for your input. 
Digging a little deeper, I think it may be as simple as launching the URL from OD with the auth credentials to solve this issue. 
@joekaufmanexpensify Once we involve a context-aware BE engineer, I also think we will have a solid way forward to resolve the issue.

**Comment by joekaufmanexpensify** [comment]
Sure, happy to figure out whether this is a bug or new feature. [Discussing](https://expensify.slack.com/archives/C07NMDKEFMH/p1733156993591979).

**Comment by joekaufmanexpensify** [comment]
Sounds like the case got missed in [the original implementation](https://github.com/Expensify/Expensify/issues/387117). We agreed in Slack we def want to handle this though.

**Comment by rojiphil** [comment]
> Sounds like the case got missed in [the original implementation](https://github.com/Expensify/Expensify/issues/387117). We agreed in Slack we def want to handle this though.

@joekaufmanexpensify Thanks for the clarification. Can we also involve an internal engineer who has context to this feature? More specifically, I am curious to know the feedback on the following comment:

> Digging a little deeper, I think it may be as simple as launching the URL from OD with the auth credentials to solve this issue.

**Comment by rojiphil** [comment]
> Just to confirm @rojiphil, this one will be an entirely backend issue?

Looks like a BE issue to me but it depends on the feedback for the comment [here](https://github.com/Expensify/App/issues/53034#issuecomment-2512138592). 
@puneetlath What do you think?

**Comment by joekaufmanexpensify** [comment]
>Looks like a BE issue to me but it depends on the feedback for the comment https://github.com/Expensify/App/issues/53034#issuecomment-2512138592.

Got it. I think any engineer can probably confirm this. Curious if @marcaaron has thoughts on whether Roji's solution is on the right track? I also asked Puneet if he has time to take a quick look in Slack too.

**Comment by puneetlath** [comment]
I agree, I think we need to have an internal engineer update the link to use the auth credentials.

**Comment by rojiphil** [comment]
Cool. Now that we have a consensus on the way ahead, I think we can make this issue as `internal`. Meanwhile, I am unassigning myself here but will stay subscribed if any FE work arises. Thanks.
cc @joekaufmanexpensify

**Comment by marcaaron** [comment]
Yeah ok I see we did this: https://github.com/Expensify/Web-Expensify/pull/41884

So we just need to ... make that a short lived auth token link instead I think and this should work. Seems kind of half implemented.

**Comment by marcaaron** [comment]
Ok repro'd and made a PR for this. Was a one line change 🥲

**Comment by joekaufmanexpensify** [comment]
Sounds good. Thanks Marc!

**Comment by joekaufmanexpensify** [comment]
Fix is out on prod. All set here. Thanks again Marc!
